### PR TITLE
[DC-2212] Remove SNOMED code descendants of 141771 and 4054924

### PR DESCRIPTION
**Pull request by ksdkalluri**

### CODE REVIEW

**Review by lrwb-aou — CHANGES_REQUESTED** [review]
Please add the new class to the `REGISTERED_TIER_DEID_CLEANING_CLASSES` and `CONTROLLED_TIER_DEID_CLEANING_CLASSES` in clean_cdr.py.

File: data_steward/cdr_cleaner/cleaning_rules/vehicular_accident_concept_suppression.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+LOGGER = logging.getLogger(__name__)
+
+SUPPRESSION_RULE_CONCEPT_TABLE = 'vehicular_accident_descendant_concepts'
+
+VEHICULAR_ACCIDENTS_CONCEPT_QUERY = JINJA_ENV.from_string("""
+CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_id}}.{{concept_suppression_lookup_table}}` AS
+SELECT
+c.*
+FROM
+  `{{project_id}}.{{dataset_id}}..concept` c
```
Comment (lrwb-aou) [inline data_steward/cdr_cleaner/cleaning_rules/vehicular_accident_concept_suppression.py:28]:
```suggestion
  `{{project_id}}.{{dataset_id}}.concept` c
```

File: data_steward/cdr_cleaner/cleaning_rules/vehicular_accident_concept_suppression.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = "Suppress Vehicular accidents related SNOMED concepts descendants."
+
+        super().__init__(
+            issue_numbers=['DC1959', 'DC2212'],
+            description=desc,
+            affected_datasets=[cdr_consts.REGISTERED_TIER_DEID],
```
Comment (lrwb-aou) [inline data_steward/cdr_cleaner/cleaning_rules/vehicular_accident_concept_suppression.py:74]: also affects controlled tier.

File: data_steward/cdr_cleaner/cleaning_rules/vehicular_accident_concept_suppression.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+# Third party imports
+from google.cloud.exceptions import GoogleCloudError
+
+LOGGER = logging.getLogger(__name__)
+
+SUPPRESSION_RULE_CONCEPT_TABLE = 'vehicular_accident_descendant_concepts'
+
+VEHICULAR_ACCIDENTS_CONCEPT_QUERY = JINJA_ENV.from_string("""
+CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_id}}.{{concept_suppression_lookup_table}}` AS
```
Comment (lrwb-aou) [inline data_steward/cdr_cleaner/cleaning_rules/vehicular_accident_concept_suppression.py:24]: I think this works as is, but the Create table docs seem to suggest enclosing the query in ().

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/vehicular_accident_concept_suppression_test.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        """
+
+        condition_insert_template = JINJA_ENV.from_string("""
+           INSERT INTO `{{fq_dataset_name}}.condition_occurrence`
+                       (condition_occurrence_id, person_id, condition_concept_id, 
+                       condition_start_date, condition_start_datetime, condition_type_concept_id, condition_source_concept_id)
+                       VALUES
+                       (100, 1, 4016214, date('2020-01-01'), timestamp('2020-01-01'), 42894222, 0),
+                       --needs to be suppressed concepts, 4054924, 4055777, 44837728 --
+                       (101, 2, 4054924, date('2020-01-01'), timestamp('2020-01-01'), 0, 0),
```
Comment (lrwb-aou) [inline tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/vehicular_accident_concept_suppression_test.py:86]: please add a record using the concept_id 44822621 in a `*_concept_id` field.  This should be dropped.

**Review by lrwb-aou — CHANGES_REQUESTED** [review]
Good turn around on this one.  Just a few suggestions.  Thanks!

**Review by lrwb-aou — APPROVED** [review]
Thank you for the changes!  This looks good.

File: data_steward/cdr_cleaner/clean_cdr.py [hunk 5]
```diff
@@ -270,6 +271,7 @@
     (FreeTextSurveyResponseSuppression,
     ),  # Should run after any data remapping rules
     (MotorVehicleAccidentSuppression,),
```
[the remark this answers is not in the document]
Comment (ksdkalluri) [reply data_steward/cdr_cleaner/clean_cdr.py:273]: we can do that, sure, we can do it post pipeline execution.

**Pull request merged**